### PR TITLE
feat(vue-app): added nuxt-loading events

### LINKED ISSUE #7893: Add event hooks to nuxt-loading component
### Is your feature request related to a problem? Please describe.

I have some async data being loaded in `fetch`. I want to scroll on the page _after_ all content is loaded. The easiest way seemed to be something like `this.$loading.$once('finished')`.

### Describe the solution you'd like

I'd like some events added to the `nuxt-loading` component so people can use it as the source of truth if stuff is still being loaded. The only one I am concerned about is `finished`, but I could see some others added as well.

### Describe alternatives you've considered

Not using the component as the single source of truth. This can be more accurate, but also takes more work. Using the `nuxt-loading` component is fast, has good support with things like axios already, and is pretty simple to use.

### PR DESCRIPTION
**Pull request by rotem-bar**
## Types of changes

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

Resolves #7893, added the following events to nuxt-loading:
- progress (on percentage change)
- started
- paused
- resumed
- failed
- finished

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [x] I have added tests to cover my changes (if not applicable, please state why)
- [x] All new and existing tests are passing.

### CODE REVIEW

**Comment by pi0** [comment]
Hi @rotem-bar and thanks for PR :heart: We have to keep it pending for next semver/minor release which is after nuxt3 api was more stable. Meanwhile you can override it by creating `app/components/nuxt-loading.vue`

**Review by Kronhyx — APPROVED** [review]

**Review by Kronhyx — APPROVED** [review]

File: packages/vue-app/template/components/nuxt-loading.vue [hunk 1]
```diff
@@ -24,6 +24,12 @@ export default {
         : (!this.reversed ? '0px' : 'auto')
     }
   },
+  watch: {
+    percent(value) {
+      if(value !== 0)
+        this.$emit('progress', value)
```
Comment (shamilfrontend) [inline packages/vue-app/template/components/nuxt-loading.vue:30]: `if(value !== 0) return;`?

**Comment by hafixo** [comment]
#7930  
Merge pull requested
packages/vue-app/template/components/nuxt-loading.vue

**Comment by danielroe** [comment]
As Nuxt 2 is currently in maintenance-mode until June 30, 2024, we are not aiming to merge any more features and so, regretfully, I am closing this PR. My apologies that we weren't able to include it in v2.17 🙏

**Pull request closed without merging**